src/System.Text.JsonLab/System/Text/Json/Utf8JsonReader.cs

```diff
@@ -620,55 +739,69 @@ private void ConsumePropertyName()
 #if UseInstrumented
             _position++;
 #endif
+            return true;
         }
 
-        private void ConsumeString()
+        private bool ConsumeString()
         {
             //Create local copy to avoid bounds checks.
             ReadOnlySpan<byte> localCopy = _buffer;
 
-            int idx = localCopy.Slice(CurrentIndex).IndexOf(JsonConstants.Quote);
+            int idx = localCopy.Slice(CurrentIndex + 1).IndexOf(JsonConstants.Quote);
```

Comment:
will the slice always succeed? i.e. are you sure the current index is not at the end? Maybe at least worth asserting.